version: '3.8'
services:
  # django
  django:
    command: python manage.py runserver 0.0.0.0:8000
    ports:
      - 80:8000
    env_file:
      - ./.env
    build:
      args:
        user: django
        uid: 1000
      context: .
      dockerfile: ./.docker/Dockerfile
    image: django-app
    restart: unless-stopped
    working_dir: /usr/src/app
    volumes:
      - ./:/usr/src/app
    networks:
      - myNetwork

  #MySQL Service
  db:
    image: mysql:5.7
    container_name: db
    restart: unless-stopped
    tty: true
    ports:
      - "3306:3306"
    environment:
      MYSQL_DATABASE: django
      MYSQL_USER: django
      MYSQL_PASSWORD: django
      MYSQL_ROOT_PASSWORD: django
    volumes:
      - mysqldata:/var/lib/mysql/
    networks:
      - myNetwork

  # phpmyadmin
  phpmyadmin:
    image: 'phpmyadmin'
    container_name: phpmyadmin
    environment:
        PMA_HOST: mysql
        PMA_PASSWORD: django
        PMA_ARBITRARY: 1
    restart: always
    ports:
        - 8080:80
    networks:
        - myNetwork

  redis:
    image: redis:6.2-alpine
    restart: always
    ports:
      - '6379:6379'
    command: redis-server --save 20 1 --loglevel warning --requirepass eYVX7EwVmmxKPCDmwMtyKVge8oLd2t81
    volumes: 
      - redis:/redis
    networks:
      - myNetwork
        
#Docker Networks
networks:
  myNetwork:
    driver: bridge
volumes:
  mysqldata:
  redis: